### PR TITLE
Update to fixed-hash 0.3.0-beta and unit 0.5.0-beta

### PR DESCRIPTION
**Pull request by Robbepop**
Updates `fixed-hash` and `uint` dependencies

> We need to merge this PR before we can update `pwasm-ethereum` or `pwasm-abi` since they are planned to no longer depend on `uint` directly but to be depending on `pwasm-std` only that provides the `uint` implementations for them.

#### Summary

This PR is going to include `pwasm-std` depending on `uint` also providing useful casts between `U256` and `H256` types so that `pwasm-ethereum` and `pwasm-abi` no longer have to depend on `uint` but just `pwasm-std`.

#### Changes

- Update `rust-toolchain` from `2018-07-24` to `2018-10-30`
- Feature `std` is now propagated to `byteorder/std`.
- Add `U256` type based on `uint` crate
- Update `fixed-hash` dependency from `0.2` to `0.3.0-beta`
- Update `uint` dependency from `0.4` to `0.5.0-beta`
- Remove all hash types but `H160` and `H256`
- `Address` is no longer a type alias but a thin wrapper around `H160` -> stronger typing
- Add conversions between `U256` and `H256`
- Add conversions between `H160` and `H256`

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,20 +13,16 @@ categories = ["no-std", "embedded"]
 exclude = [ "demo/*" ]
 
 [dependencies]
+fixed-hash = { version = "0.3.0-beta.0", default-features = false }
```
Comment (dvdplm) [inline Cargo.toml]: I think you can use `version = "0.3.0-beta"` here. Saves some time if there are other pre-releases.

Comment (Robbepop) [reply Cargo.toml]: ah that's good to know! thanks. :)

Comment (Robbepop) [reply Cargo.toml]: done

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "pwasm-std"
-version = "0.10.0"
+version = "0.11.0-beta.0"
```
Comment (NikVolf) [inline Cargo.toml]: please don't update any metadata in the pr itself, it will be separate commit

**Comment by NikVolf** [comment]
is it still WiP, no?

File: Cargo.toml [hunk 3]
```diff
@@ -13,20 +13,16 @@ categories = ["no-std", "embedded"]
...
-
-[dependencies.parity-hash]
-version = "1"
-default-features = false
+static_assertions = "0.2"
 
 [features]
-default = []
-std = ["parity-hash/std"]
+default = ["std"]
```
Comment (NikVolf) [inline Cargo.toml]: it should not be default

pwasm-std is designed to replace std, not to come with default std, so this feature is activated only in some testing scenarios

File: Cargo.toml (hunk 2 above)
Comment (Robbepop) [reply Cargo.toml]: done

File: Cargo.toml (hunk 3 above)
Comment (NikVolf) [reply Cargo.toml]: Yeah, but many other crates don’t aim to replace std by default :) so yeah. I’m against

Comment (Robbepop) [reply Cargo.toml]: okay will revert this then

Comment (Robbepop) [reply Cargo.toml]: Do you have an idea how we could solve the problems with doc-tests for this crate?

Comment (Robbepop) [reply Cargo.toml]: > okay will revert this then

done

File: src/crypto.rs [hunk 4]
```diff
@@ -1,28 +1,31 @@
...
 /// 	3, 187, 250, 37, 76, 176, 31, 173
-/// ];
+/// ]);
 ///
-/// assert_eq!(hash.as_ref(), expected);
+/// assert_eq!(hash, expected);
 /// ```
-pub fn keccak<T>(input: T) -> H256 where T: AsRef<[u8]> {
+pub fn keccak<T>(input: T) -> H256
+where
```
Comment (fckt) [inline src/crypto.rs]: hmm.. I wonder, is that ok indentation?..

Comment (Robbepop) [reply src/crypto.rs]: It is standard Rust indentation convention but maybe against parity rules - I have to check that.

**Comment by Robbepop** [comment]
Anybody has got an idea why the CI is failing?
It is not only for this PR but also for the master without all the work in this PR.

**Comment by Robbepop** [comment]
I think the CI is failing because of some `dev-dependencies` of the `uint` crate.
I have fixed this in [another PR](https://github.com/paritytech/parity-common/pull/77) and hopefully it will fix this CI pipeline, too.

**Comment by Robbepop** [comment]
Blocked by https://github.com/paritytech/parity-common/pull/77

**Comment by Robbepop** [comment]
This PR is ready for merging.

**Comment by Robbepop** [comment]
> lgtm!
Would be nice to make sure also if the dependent crates doesn't break with this refactoring (if it's not checked yet).

@fckt Dependend crates will most certainly break due to the changes introduced in the PR but I am fixing them as soon as I am able to release a new beta version of this crate.
